awx/ui/test/e2e/tests/test-applications.js

```diff
@@ -74,6 +90,61 @@ module.exports = {
             .waitForElementVisible('@description')
             .expect.element('@description').value.to.equal(store.application.description);
     },
+    'add a read scoped application token': client => {
+        client.logout();
+        client.login(store.adminUser.username, store.adminUser.password);
+
+        // Given the application created on the previous test
+        const token = {
+            application: store.application.name,
+            description: `Read scoped token for ${store.application.name}`,
+            scope: 'Read',
+        };
+        const users = client.page.users();
+        users.load();
+        client.waitForSpinny();
+        users.createToken(store.adminUser.username, token, (tokenInfo) => {
+            const applications = client.page.applications();
+            applications.load();
+            client.waitForSpinny();
+            applications.search(store.application.name);
+            client.findThenClick(editLink, 'css');
+            applications.section.edit
+                .waitForElementVisible('@tokensTab')
+                .click('@tokensTab')
+                .waitForSpinny();
+            // FIXME: Search is not properly working, update this when it is working.
+            applications.section.tokens.expect.element('@list').text.to.contain(`${store.adminUser.username}\n${token.description}\nEXPIRATION ${tokenInfo.expires}`);
+        });
+    },
+    'add a write scoped application token': client => {
+        // Given the application created on the previous test
+        const token = {
+            application: store.application.name,
+            description: `Write scoped token for ${store.application.name}`,
+            scope: 'Write',
+        };
+
+        const users = client.page.users();
+        users.load();
+        client.waitForSpinny();
+        users.createToken(store.adminUser.username, token, (tokenInfo) => {
+            const applications = client.page.applications();
+            applications.load();
+            client.waitForSpinny();
+            applications.search(store.application.name);
+            client.findThenClick(editLink, 'css');
+            applications.section.edit
+                .waitForElementVisible('@tokensTab')
+                .click('@tokensTab')
+                .waitForSpinny();
+            // FIXME: Search is not properly working, update this when it is working.
+            applications.section.tokens.expect.element('@list').text.to.contain(`${store.adminUser.username}\n${token.description}\nEXPIRATION ${tokenInfo.expires}`);
+        });
+
+        client.logout();
+        client.login();
+    },
     'delete an application': client => {
```

Comment:
do we want to delete both tokens here?

Reply:
@unlikelyzero we could but I assume deleting the app (which is done by the last test) would also remove them. But if you are talking about having that as extra coverage I can add it no problem